examples/07_multiple_key_join.py

```diff
@@ -0,0 +1,154 @@
+#       Here, we consider only weather measurements from 2008.
+
+weather = pd.read_parquet("https://figshare.com/ndownloader/files/41771457")
+# Sampling for faster computation.
+weather = weather.sample(100_000, random_state=1, ignore_index=True)
+weather.head()
+
+###############################################################################
+# Joining
+# -------
+# First we join the stations with weather on the ID (exact join):
+
+aux = pd.merge(stations, weather, on="ID")
+aux.head()
+
+###############################################################################
+# Then we join this table with the airports so that we get all auxilliary
+# tables into one.
+
+from skrub import Joiner
+
+joiner = Joiner(
+    tables=[(airports, ["lat", "long"])], main_key=["LATITUDE", "LONGITUDE"]
+)
+
+aux_augmented = joiner.fit_transform(aux)
+
+aux_augmented.head()
+
+###############################################################################
+# Joining airports with flights data:
+# Another multiple key join on the date and the airport
+
+joiner = Joiner(
+    tables=[(aux_augmented, ["YEAR/MONTH/DAY", "iata"])],
+    main_key=["Year_Month_DayofMonth", "Origin"],
+)
+
+main = joiner.fit_transform(flights)
+main.head()
+
+###############################################################################
+# We now have combined all the information from our pool of tables into one.
+# We will use this main table to model the prediction of flight delay.
+# Training data is passed through a |Pipeline|:
+
+
+from skrub import TableVectorizer
+from sklearn.ensemble import HistGradientBoostingClassifier
+from sklearn.pipeline import make_pipeline
+
+tv = TableVectorizer()
+hgb = HistGradientBoostingClassifier()
+
+pipeline_hgb = make_pipeline(tv, hgb)
+
+###############################################################################
+# We isolate our target variable and remove useless ID variables:
+
+y = main["ArrDelay"]
+X = main.drop(columns=["ArrDelay", "FlightNum", "TailNum", "ID", "iata"])
+
+###############################################################################
+# We want to frame this as a classification problem:
+# suppose that your company is obliged to reimburse the ticket
+# price if the flight is delayed.
+# We have a binary classification problem:
+# the flight was delayed (1) or not (0).
+
+y = (y > 0).astype(int)
+y.value_counts()
+
+###############################################################################
+# The results:
+
+from sklearn.model_selection import cross_val_score
+
+scores = cross_val_score(pipeline_hgb, X, y)
+scores.mean()
+
+###############################################################################
+# Conclusion
+# ----------
+#
+# In this example, we have combined multiple tables with complex joins
+# on imprecise and multiple-key correspondences.
+# This is made easy by skrub's |Joiner| transformer.
+#
+# Our final cross-validated accuracy score is 0.6.
```

Comment:
Yes, it's mostly the GapEncoder. Especially when working on the ID columns, which is why I dropped them (#585).
I hope this will get much better after speedups #680